File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int[] ans = new int[nums.size()];
        for(int i=0;i<nums.size();i++)
        {
            ans[i] = findValue(nums.get(i));
        }
        return ans;
    }
    private int findValue(int n)
    {
        for(int i=1;i<n;i++)
        {
            int sum = i | (i+1);
            if(sum==n)
            {
                return i;
            }
        }
        return -1;
    }
}

File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        int tem;
        vector<int>result;
        for(int t : nums){
            int tem =-1;
            for(int i = 1; i<=t;i++){
                if((i | (i+1)) == t)
                {
                    tem=i;
                    break;
                }
            }
            result.push_back(tem);    
        }
        return result;      
    }
};